### PR TITLE
Feature/design system update

### PR DESCRIPTION
**Pull request by davidsauntson**
Updates the design system to 4.0.3 and:
 - refactors landing menu styles into dedicated file and BEM
 - corrects link as button interaction colours (this should really just be a link)
 - add required design-system grid structure
 - refactor menu tiles to improve accessibility
 - add skip links
 - removes `/intranet` as url prefix so works OOB (will address in subsequent PR)

NB this does not include replacing images with SVGs or narrowing SCSS imports

### CODE REVIEW

File: assets/stylesheets/index.scss [hunk 1]
```diff
@@ -1,39 +1,43 @@
...
-  li {
-    width: 120px;
-    list-style-type: none;
-    text-align: center;
-    border: 1px transparent solid;
-    padding: 5px;
+.logo {
+  margin-top: $cads-spacing-7;
+  display: block;
+  background: url("/source/images/logo.png") no-repeat;
```
Comment (timwis) [inline assets/stylesheets/index.scss:8]: Have you tested this in the built version? Just looking at this line, I would expect the built page to request this from `/intranet/source/images....`, which would be 404. Unless webpack knows to pull that out?

File: source/index.html.md.erb [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-</a>
+<div class="cads-page-content">
+  <div class="cads-grid-container">
+    <div class="cads-grid-row">
+      <main class="cads-grid-col-12">
+        <h1 class="cads-page-title">Your digital workplace</h1>
+        <%= partial "partials/menu" %>
+        <div class="cads-prose">
+          <h2>A new intranet</h2>
+          <p>
```
Comment (timwis) [inline source/index.html.md.erb]: FYI you can change the filename back to `.html.erb` if we're not using markdown here anymore, if you want

File: config.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@
 page '/*.json', layout: false
 page '/*.txt', layout: false
 
-config[:http_prefix] = '/intranet'
+config[:http_prefix] = ''
```
Comment (timwis) [inline config.rb]: When I tried it without this, it no longer loaded the css file in production, as it was being requested from `citizensadvice.github.io/assets/stylesheets...`

**Review by timwis — COMMENTED** [review]
Looks great! Left a couple comments about things that I wouldn't have expected to work in the built/prod version, but assuming you've tested it, then feel free to disregard!

File: assets/stylesheets/index.scss (hunk 1 above)
Comment (davidsauntson) [reply assets/stylesheets/index.scss:8]: Looking at the build CSS, webpack is converting the image into `data:img` so there isn't a request to the actual path, think it'll be OK

**Pull request merged**